edsdme/scripts/personalization.js

```diff
@@ -181,6 +197,68 @@ function personalizeDropdownElements(profile) {
   });
 }
 
+function personalizeMainNavMenu(item) {
+  // section separators
+  const separatorSelector = 'h5';
+  const separators = item.querySelectorAll(separatorSelector);
+  let elements = Array.from(separators).filter(item => {
+    return item.textContent.includes(PERSONALIZATION_MARKER);
+  });
+  const processedSeparators = processGnavElements(elements);
+  processedSeparators.forEach(({ el, conditions }) => {
+    if (!el || !conditions) return;
+    const nextElementSibling = el.nextElementSibling;
+    if (nextElementSibling?.tagName !== separatorSelector && shouldHide(conditions)) {
+      nextElementSibling.remove();
+    }
+    hideElement(el, conditions, true);
+  });
+
+  // links
+  const links = item.querySelectorAll('a');
+  elements = Array.from(links).filter(item => {
+    return item.textContent.includes(PERSONALIZATION_MARKER);
+  });
+  const processedLinks = processGnavElements(elements);
+  processedLinks.forEach(({ el, conditions }) => {
+    if (!el || !conditions) return;
+    const listItem = el.closest('li');
+    hideElement(listItem || el, conditions, true);
+  });
+
+  // link group blocks
+  const linkGroups = item.querySelectorAll('.link-group');
+  elements = Array.from(linkGroups).filter(item => {
+    return item.className.includes(PERSONALIZATION_MARKER);
+  });
+  elements.forEach((el) => {
+    const conditions = Object.values(el.classList);
+    hideElement(el, conditions, true);
+  });
+}
+
+function personalizeMainNav(gnav) {
```

Comment:
Maybe this can be done in a simpler way: 
- Profile block is the last block in the gnav (and we can say in the documentation that it needs to be the last block authored)
```
function personalizeMainNav(gnav) {
  let profile;
  // Remove profile because it will be personalized in the next step
  if (gnav.querySelector('.profile')) {
    const lastElement = gnav.lastElementChild;
    profile = gnav.removeChild(lastElement);
  }
  // Now in the gnav we only have main nav elements
  const personalizationXPath = `//*[contains(text(), "${PERSONALIZATION_MARKER}")]`;
  const elements = getNodesByXPath(personalizationXPath, gnav);
  // All elements with text containing 'partner-personalization'
  // Links, headers, text etc.
  const processedElements = processGnavElements(elements);
  processedElements.forEach(({el, conditions}) => {
    // Hiding logic
    if (!el || !conditions) return;
    hideElement(el, conditions, true);
  });
  const linkGroups = gnav.querySelectorAll('.link-group.partner-personalization');
  Array.from(linkGroups).forEach((link) => {
    //Hiding logic for link groups
  });
  
  // Return profile into the gnav so it can be personalized in the next step
  profile && gnav.appendChild(profile);
}
```
And it looks like to me that the only thing we need to worry about is dropdown with columns, we can check if element is `h5` and does it have next sibling `ul` and remove both (you did something similar), for the rest of the elements I think we are okay with removing the parent of the element that needs to be hidden.

Reply:
Thanks @zagi25 for the suggestion, updated in [22c17c8](https://github.com/adobecom/dme-partners/pull/57/commits/22c17c81627eba5271880d111dc9e0b39e93e016), see [personalizeMainNav](https://github.com/adobecom/dme-partners/pull/57/files#diff-2b941a13cbb23e57ba4f11aa2f63da3536ce8ce22feec620a96b7b2eaebcaa37R198)

One thing - we can't arbitrarily remove the parent of the matched items since some are wrapped and others not (e.g. `<h2><a>my link(partner-personalization, partner-level-gold)</a></h2>` versus `<h2>my text(partner-personalization, partner-level-gold)</h2>`), so I remove wrappers only if they are a `h2` or `li`